src/Microsoft.Azure.SignalR/Startup/ServiceHubDispatcher.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#if NETCOREAPP3_0
+using System;
+using Microsoft.AspNetCore.Connections;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.Azure.SignalR.Startup
+{
+    internal class ServiceHubDispatcher
+    {
+        // A late bound version of ServiceHubDispatcher<THub>.
+        private readonly Type _serviceDispatcherType = typeof(ServiceOptions).Assembly.GetType("Microsoft.Azure.SignalR.ServiceHubDispatcher`1");
```

Comment:
Sine we're inside of the same assembly, this could be a bit cleaner: 

```C#
private readonly Type _serviceDispatcherType = typeof(ServiceHubDispatcher<>);
```